compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -333,6 +333,198 @@ bool OMR::CFGSimplifier::simplifyInstanceOfTestToCheckcast(bool needToDuplicateT
    return true;
    }
 
+bool OMR::CFGSimplifier::simplifyBoundCheckWithThrowException(bool needToDuplicateTree) 
```

Comment:
It would be great to have some kind of diagram here of the tree/cfg patterns before and after for understandability.

Reply:
Added the explanation of what the code is trying to do.